src/core/Ros.js

```diff
@@ -624,5 +625,37 @@ Ros.prototype.decodeTypeDefs = function(defs) {
   return decodeTypeDefsRec(defs[0], defs);
 };
 
+/**
+ * Retrieves list of topics and their associated type definitions.
+ *
+ * @param callback function with params:
+ *   * topics - Array of topic names
+ *   * types - Array of message type names
+ *   * typedefs_full_text - Array of full definitions of message types, similar to `gendeps --cat`
```

Comment:
Would you mind documenting `failedCallback` here?

Reply:
Yeah this was missing throughout this file. I added a commit that adds this all over the file (based on https://github.com/RobotWebTools/roslibjs/blob/c10d9eca43ad51cc89b335d00e81f647cbcb77d3/src/core/Service.js#L36) and also fixes some other documentation strings in minor ways here and there.